compiler/rustc_mir_transform/src/instcombine.rs

```diff
@@ -142,6 +143,14 @@ impl<'tcx> InstCombineContext<'tcx, '_> {
         }
     }
 
+    fn combine_cast(&self, _source_info: &SourceInfo, rvalue: &mut Rvalue<'tcx>) {
+        if let Rvalue::Cast(_kind, operand, ty) = rvalue {
+            if operand.ty(self.local_decls, self.tcx) == *ty {
```

Comment:
Are regions erased in the MIR body here? Otherwise this won't apply any time a lifetime is present in the type being casted.

Reply:
Yes, they should be. Seems like a good candidate for a test case though

Reply:
Hm. I made an attempt, but I do not think it is optimal. Probably good enough though?